test/issue806.test.js

```diff
@@ -0,0 +1,35 @@
+import React from "react"
+import { configure, observable } from "mobx"
+import { observer } from "../src"
+import { render } from "@testing-library/react"
+import withConsole from "./utils/withConsole";
+
+
+@observer
+class Issue806Component extends React.Component {
+    render() {
+        return null;
+    }
+}
+
+test("verify issue 806", () => {
+    configure({
+        observableRequiresReaction: true
+    });
+
+    const x = observable({
+        a: 1
+    });
+
+    withConsole(["warn"], () => {
+        render(<Issue806Component />);
```

Comment:
I'd expand the test case to where one observer component passes some of it's observables to another observer component